src/components/CreateAnalysis/CreateAnalysisDetail.tsx

```diff
@@ -1,22 +1,29 @@
-import React, { Dispatch, SetStateAction, useContext, useEffect } from "react";
+import React, { Dispatch, SetStateAction, useContext, useEffect, useState } from "react";
 import avator from '../../assets/images/avator.png';
 import { CreateAnalysisTypes } from "../../context/actions/types";
 import { AppContext } from "../../context/context";
 import RightArrowButton from "../../pages/CreateAnalysisPage/RightArrowButton";
 import CreateAnalysisService, { StudyInstance } from "../../services/CreateAnalysisService";
+import ModelSelection from "./ModelSelection";
 import SelectedStudyDetail from "./SelectedStudyDetail";
 import SelectionStudy from "./SelectionStudy";
+import plugins from '../../api/app.config'
 
 
 interface CreateAnalysisDetailProps {
   setIsExpanded: Dispatch<SetStateAction<boolean>>,
-  submitAnalysis: () => void
+  submitAnalysis: (XrayModel: string | void, CTModel: string | void) => (() => void)
 }
 
 const CreateAnalysisDetail: React.FC<CreateAnalysisDetailProps> = (props) => {
   const { state: { dcmImages, createAnalysis }, dispatch } = useContext(AppContext);
   const { selectedStudyUIDs } = createAnalysis;
 
+  const [XrayModel, setXrayModel] = useState(Object.keys(plugins.XrayModels)[0]);
+  const [CTModel, setCTModel] = useState(Object.keys(plugins.CTModels)[0]);
+  const [modelValue, setModelValue] = useState(0);
```

Comment:
As you mentioned during our meeting, the SelectionStudy list isn't limited to just 1 XRay study and 1 CT study (for example, in the screenshot below, it's possible to have 1 study with CT scans, and 2 with XRay scans). I think we may be able to address this issue with the following changes:

Instead of using indices to track which study is currently selected and modify `useXray` in a binary /toggle fashion, I would consider using a string to store the actual Modality of the currently selected study. In the `SelectionStudy` component, we already have information on what Modality it contains via its props. We could modify the `setModelType` function to take in a string of the Modality instead of a number, and update `modelValue`. Then, `useXray` can be calculated as `modelValue === 'CR'`.

```suggestion
const [modelValue, setModelValue] = useState('');
```
![Screenshot from 2021-02-08 16-34-48](https://user-images.githubusercontent.com/24356368/107284310-98862c00-6a2b-11eb-9512-1998ff2b3119.png)

Reply:
Agreeing with the comment. There seems to a lot of dependency with indices with studies limited to CTModels and XrayModels.

I would suggest discussing with Ivan to decide whether its something we want to take care of now or wait when we actually have PACS integrated so we can see how the real data differs from the mocked data.

Reply:
Great suggestion, helps for future flexibility for changes. Regarding issue of getting the Modality value of the study that is initially selected, the code now dynamically checks to see the type of study that is initially selected (and shows the corresponding drop-down)